rust/connlib/tunnel/src/io.rs

```diff
@@ -581,14 +669,62 @@ mod tests {
         assert!(timeout >= now, "timeout = {timeout:?}, now = {now:?}");
     }
 
+    #[tokio::test]
```

Comment:
perhaps it would be worth adding a test that we would re-bootstrap after `io.reset()`?

Reply:
I think the main behaviour of bootstrapping is covered with this test, there is little risk in that re-bootstrapping behaviour to be wrong and I'd like to avoid as many of these IO tests as possible because they can easily be flaky.